installers/BuildFromSource.hs

```diff
@@ -127,22 +127,32 @@ makeRepos artifactDirectory repos =
  do createDirectoryIfMissing True artifactDirectory
     setCurrentDirectory artifactDirectory
     root <- getCurrentDirectory
-    cabal [ "sandbox", "init" ]
     mapM_ (uncurry (makeRepo root)) repos
 
+    -- create a sandbox for installation
+    cabal [ "sandbox", "init" ]
+
+    -- add each of the sub-directories as a sandbox source
+    cabal ([ "sandbox", "add-source" ] ++ map fst repos)
+    
+    -- install all of the packages together in order to resolve transitive dependencies robustly
+    -- (install the dependencies a bit more quietly than the elm packages)
+    cabal ([ "install", "-j", "--only-dependencies", "--ghc-options=\"-w\"" ] ++ map fst repos)
+    cabal ([ "install", "-j", "--bindir=../bin", "--ghc-options=\"-XFlexibleContexts\"" ] ++ filter (/= "elm-reactor") (map fst repos))
```

Comment:
I suspect that's the reason. I've needed `-XFlexibleContexts` for a while now when building some part of Elm on my laptop (GHC 7.10, because Arch bleeding edge).